openslides_backend/action/mixins/import_mixins.py

```diff
@@ -1,27 +1,147 @@
 
 
 class ImportState(str, Enum):
-    ERROR = "error"
-    NEW = "new"
+    NONE = "none"
     WARNING = "warning"
+    NEW = "new"
     DONE = "done"
     GENERATED = "generated"
+    REMOVE = "remove"
+    ERROR = "error"
+
+
+class ResultType(Enum):
+    """Used by Lookup to differ the possible results in check_duplicate."""
+
+    FOUND_ID = 1
+    FOUND_MORE_IDS = 2
+    NOT_FOUND = 3
+
+
+class Lookup:
+    def __init__(
+        self,
+        datastore: DatastoreService,
+        collection: str,
+        name_entries: List[Tuple[Union[str, Tuple[str, ...]], Dict[str, Any]]],
+        field: Union[str, Tuple[str, ...]] = "name",
+        mapped_fields: Optional[List[str]] = None,
+    ) -> None:
+        if mapped_fields is None:
+            mapped_fields = []
+        self.datastore = datastore
+        self.collection = collection
+        self.field = field
+        self.name_to_ids: Dict[
+            Union[str, Tuple[str, ...]], List[Dict[str, Any]]
+        ] = defaultdict(list)
+        for name, _ in name_entries:
+            self.name_to_ids[name] = []
+        self.id_to_name: Dict[int, List[Union[str, Tuple[str, ...]]]] = defaultdict(
+            list
+        )
+        or_filters: List[Filter] = []
+        if "id" not in mapped_fields:
+            mapped_fields.append("id")
+        if type(field) == str:
+            if field not in mapped_fields:
+                mapped_fields.append(field)
+            if name_entries:
+                or_filters = [
+                    FilterOperator(field, "=", name) for name, _ in name_entries
+                ]
+        else:
+            mapped_fields.extend((f for f in field if f not in mapped_fields))
+            if name_entries:
+                or_filters = [
+                    And(*[FilterOperator(field[i], "=", name_tpl[i]) for i in range(3)])
+                    for name_tpl, _ in name_entries
+                ]
+        if or_filters:
+            for entry in datastore.filter(
+                collection,
+                Or(*or_filters),
+                mapped_fields,
+                lock_result=False,
+            ).values():
+                self.add_item(entry)
+
+        # Add action data items not found in database to lookup dict
+        for name, entry in name_entries:
+            if not (values := cast(list, self.name_to_ids[name])):
+                values.append(entry)
+            else:
+                if not values[0].get("id"):
+                    values.append(entry)
+
+    def check_duplicate(self, name: Union[str, Tuple[str, ...]]) -> ResultType:
+        if len(values := self.name_to_ids.get(name, [])) == 1:
+            if values[0].get("id"):
+                return ResultType.FOUND_ID
+            else:
+                return ResultType.NOT_FOUND
+        elif len(values) > 1:
+            return ResultType.FOUND_MORE_IDS
+        raise ActionException("Logical Error in Lookup Class")
+
+    def get_x_by_name(
```

Comment:
```suggestion
    def get_field_by_name(
```
x does not convey any meaning of the intended function of the method

Reply:
done